### PR TITLE
Updated Ownership Interest Allocations

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#14714 /bcgov/entity#14805

*Description of changes:*
* Removed the process by which denominator was calculated and auto applied to each individual group.
* Refactored the means by which the fractions were added to handle mismatched fractions.
* Reduce fractions to simplest lowest terms.  
* Updated validation and error messaging.

Screen Shot 2023-01-13 at 1 45 00 PM
Screen Shot 2023-01-13 at 1 44 42 PM
Screen Shot 2023-01-13 at 1 44 31 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by cameron-eyds** [comment]
When reviewing, utilize the 'hide white space' for easier reviewing 👍 
Screen Shot 2023-01-13 at 1 49 23 PM

File: ppr-ui/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ppr-ui",
-  "version": "0.3.78",
+  "version": "0.4.0",
```
Comment (cameron-eyds) [inline ppr-ui/package.json:3]: With the upcoming sbc pilot and eventual release of Search, thought it was time to move up.

Comment (dimak1) [reply ppr-ui/package.json:3]: Good, I think this was a long overdue.

**Review by dimak1 — APPROVED** [review]
Looks good, especially maths calculations 😄 👍🏻

**Comment by dimak1** [comment]
> When reviewing, utilize the 'hide white space' for easier reviewing 👍 Screen Shot 2023-01-13 at 1 49 23 PM

I also like Split view (because I can see the new code all at once).

**Comment by cameron-eyds** [comment]
> > When reviewing, utilize the 'hide white space' for easier reviewing 👍 Screen Shot 2023-01-13 at 1 49 23 PM
> 
> I also like Split view (because I can see the new code all at once).

@dimak1 Oh right on! 
I haven't tried this but i will now for sure, thanks!

**Review by ChasBoucher — APPROVED** [review]

**Review by ChasBoucher — APPROVED** [review]
Looks good.

**Pull request merged**